kura/org.eclipse.kura.deployment.agent/src/main/java/org/eclipse/kura/deployment/agent/impl/DeploymentAgent.java

```diff
@@ -466,16 +453,18 @@ private void removePackageFromConfFile(String packageName) {
         oldDeployedPackages.putAll(deployedPackages);
         deployedPackages.remove(packageName);
 
-        if (this.dpaConfPath == null) {
-            logger.warn("Configuration file not specified");
-            return;
+        if (!oldDeployedPackages.equals(deployedPackages)) {
+            writeDPAPropertiesFile(this.dpaConfPath, deployedPackages);
         }
+    }
 
-        if (oldDeployedPackages.equals(deployedPackages)) {
+    private static void writeDPAPropertiesFile(String dpaConfPath, Properties deployedPackages) {
+        if (dpaConfPath == null) {
```

Comment:
Can `this.dpaConfPath` be null?

Reply:
Don't know. I just refactored the existing code extracting the duplicated lines. I can dig into it but I prefer leaving it as-is...

Reply:
I would dig deeper, we unlikely will have the chance to refactor this again...

Reply:
Soooooo... a few lines above we have [this interesting check](https://github.com/mattdibi/kura/blob/f5165dfd6f0a8884706879f5a939ec60067db1e0/kura/org.eclipse.kura.deployment.agent/src/main/java/org/eclipse/kura/deployment/agent/impl/DeploymentAgent.java#L124-L127)...

I might need a little help here but I'll try to reconstruct what can happen: when the `activate` method is run and there's no dpa properties file path specified an exception is thrown. This means that the class will not publish its services (as it failed to initialise) and thus the code for the `writeDPAPropertiesFile` will not be ran with a null `dpaConfPath`... right?

Reply:
Correct, also at this point the check becomes useless: https://github.com/eclipse/kura/blob/0c712b5746302df50bfe9d29a250d90877525653/kura/org.eclipse.kura.deployment.agent/src/main/java/org/eclipse/kura/deployment/agent/impl/DeploymentAgent.java#L351